Login UI improv liu wu

WeMWish opened the pull request.

Dsuzuki126 commented:
A few things:
1) I recommend you guys write jest tests as manually testing is difficult and I dont think you wrote any (but if you did please show me where)
2) the password reset doesnt have any effect unless I copy and paste the url for the login page into another tab. Refreshing the page and clearing the cache doesn't work. 
3) when I click the "confirm" button there is still a prompt that says incorrect password 
4) the indexDB information doesnt reflect the change in real time hence having to go to another tab
5) small thing, in LoginJS.js on line 98 there is a 'g' missing in the console.log that may cause issues

Dsuzuki126 commented:
Another thing I discovered is that if I copy paste the url to another tab then go to a previous tab that I reset the password with and hard refresh the page then I can use the new PIN but the indexDB information is still wrong on the previous tab

Dsuzuki126 reviewed (changes requested):
A few things:
Thanks for removing the unnecessary Login alert
- I recommend you guys write jest tests as manually testing is difficult and I dont think you wrote any (but if you did please show me where); i think if you ran through the functionality you can find different errors that come up
- The password reset doesnt have any effect unless I copy and paste the url for the login page into another tab. Refreshing the page and clearing the cache doesn't work.
- when I click the "confirm" button there is still a prompt that says incorrect password
- the indexDB information doesnt reflect the change in real time hence having to go to another tab
- small thing, in LoginJS.js on line 98 there is a 'g' missing in the console.log that may cause issues
- the way you have implemented the password reset also makes it possible for a user to reset their password PIN to an empty string
- Another thing I discovered is that if I copy paste the url to another tab then go to a previous tab that I reset the password with and hard refresh the page then I can use the new PIN but the indexDB information is still wrong on the previous tab

WeMWish commented:
- reset button is now fixed
- the incorrect password message is gone now when clicking confirm button
- line 98, console.log fixed
- the check password functionality is handled by other developers, and since the original code does include this functionality, we decided to leave it
- indexedDB should change dynamically now
- I added a jest test, but it fail to select the reset-password button on the login page. Since learning the jest test takes longer time than I expected and the buttons' functionality is pretty intuitive, to avoid affecting the next iteration, I would leave the test for now.

Dsuzuki126 commented:
Thanks for making the changes. 
I apologize as I think some of the indexDB not updating realtime may have been on my side.
There is one thing I am still worried about is that you can still click reset PIN and confirm with no PIN inputted so they could be entering the site with a PIN of empty string so maybe if you could check to make sure that the PIN isn't empty

WeMWish commented:
the check for empty password functionality is updated. Thanks for the review!

Dsuzuki126 commented:
Noice I think its good now. Thanks guys

k9kathyli commented on package-lock.json:
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "cse110-sp21-group12",
```
what are the changes in is this file for?

k9kathyli commented on source/Login/LoginJS.js:
```diff
@@ -84,11 +92,49 @@ function handleSignup(newUsername, newPassword) {
     goHome();
 }
 
+/**
+ * handle reset password functionaliy of the associated
+ */
+/**
+ * handle the login button functionalities
+ */
```
weird commenting here

k9kathyli commented on source/Login/LoginJS.js:
```diff
@@ -84,11 +92,49 @@ function handleSignup(newUsername, newPassword) {
+                sessionStorage.setItem('loggedIn', 'true');
+                goHome();
+                // // reset the button after clicked and update the settings
+                // loginButton.innerHTML = 'Sign-In';
+                // loginButton.addEventListener('click', handleLoginButton);
```
Commented code needed? If not, remove

k9kathyli commented on source/tests/PuppTests.test.js:
```diff
@@ -874,4 +874,30 @@ describe('basic navigation for BJ', () => {
+
+        const url2 = await page.evaluate(() => location.href);
+        expect(url2).toMatch(
+            'https://cse110-sp21-group12.github.io/cse110-sp21-group12/source/Index/Index.html'
```
wrong url

k9kathyli commented on source/tests/PuppTests.test.js:
```diff
@@ -874,4 +874,30 @@ describe('basic navigation for BJ', () => {
+        jest.setTimeout(30000);
+
+        await page.goto(
+            'https://cse110-sp21-group12.github.io/cse110-sp21-group12/source/Login/Login.html'
```
wrong url

k9kathyli reviewed (changes requested):
- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [x] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions 
- [ ] Code is commented/documented well
Please test locally and upload proof as discussed in Slack.

WeMWish commented:
fix puppet test, style issue fixed

PatriciaWuu commented:
I also removed all commented code as resolve #15 in this branch to wait for PR

PatriciaWuu commented:
<img width="391" alt="Screen Shot 2022-05-15 at 20 10 10" src="https://user-images.githubusercontent.com/49573110/168692295-05d19520-9892-4062-ab10-1357b76b1e87.png">

PatriciaWuu commented:
<img width="677" alt="Screen Shot 2022-05-16 at 15 48 47" src="https://user-images.githubusercontent.com/49573110/168694753-c0aaa546-35cf-459d-bc59-2dd803f3ead5.png">

Dsuzuki126 reviewed (changes requested):
make sure you pull all the changes from the main branch;

Dsuzuki126 commented:
just a reminder that after pulling from main, you may have conflicts and need to choose which code to use.  Make sure to resolve all change conflicts. It seems like the pupptests.test.js file some of your code mixed with the main branch code

Dsuzuki126 commented on source/DailyOverview/DailyOverview.html:
```diff
@@ -10,8 +10,13 @@
         <div id="header">
             <h1 id="monthView">
-                <a href="../MonthlyOverview/MonthlyOverview.html" id="words">
-                </a>
```
You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

Dsuzuki126 commented on source/MonthlyOverview/MonthlyOverview.html:
```diff
@@ -11,7 +11,12 @@
+                    <a
+                        href="../YearlyOverview/YearlyOverview.html"
+                        style="text-decoration: none"
+                        id="back"
```
You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

Dsuzuki126 commented on source/YearlyOverview/YearlyOverview.html:
```diff
@@ -10,7 +10,13 @@
+                <a
+                    href="../Index/Index.html"
+                    style="text-decoration: none"
+                    id="back"
```
You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

Dsuzuki126 reviewed (changes requested):
just a reminder that after pulling from main, you may have conflicts and need to choose which code to use. Make sure to resolve all change conflicts. It seems like the pupptests.test.js file some of your code mixed with the main branch code

- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [x] Adequate test coverage
- [x]  Follows coding conventions
- [x] Code is commented/documented well

PatriciaWuu reviewed (approved):
resolved

Dsuzuki126 commented on source/tests/PuppTests.test.js:
```diff
@@ -54,7 +54,7 @@ describe('basic navigation for BJ', () => {
         await page.waitForTimeout(1000);
     });
 
-    it('Test1: Initial Home Page - Shows create your login ', async () => {
```
I tried running the Puppeteer tests with your code and it failed a lot so maybe you can take time to figure out what is going on or just send a screenshot that it works on your machine
![image](https://user-images.githubusercontent.com/76570554/171767834-bce60cfe-8454-40ae-86e3-d4b24be1efc3.png)

Dsuzuki126 reviewed (changes requested):
- [ ] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions
- [ ] Code is commented/documented well

Dsuzuki126 reviewed (approved):
I ran the tests on my computer and they all passed. Just make sure they pass for you guys.
- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well

Dsuzuki126 reviewed (approved):
Doesn't look like any of the files changed for me. 
All the tests passed as well.

- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well

ManitKSoni reviewed (approved):
- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions
- [x] Code is commented/documented well

Dsuzuki126 reviewed (approved):
Nice job fixing the double error thing
- [ ] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well
